include/mixer.h

```diff
@@ -54,8 +55,19 @@ enum MixerModes {
 
 // A simple stereo audio frame
 struct AudioFrame {
-	float left = 0;
+	float left  = 0;
 	float right = 0;
+
+	float &operator[](std::size_t i)
+	{
+		assert(i < 2);
+		return i == 0 ? left : right;
+	}
+	const float &operator[](std::size_t i) const
+	{
+		assert(i < 2);
+		return i == 0 ? left : right;
+	}
```

Comment:
Suggest moving to `constexpr`, which will let the compiler evaluate the content at compile compile when we're passing in known literal values (especially useful for all that [0] and [1] array-access code)

``` c++
	constexpr float &operator[](const size_t i) noexcept
	{
		assert(i < 2);
		return i == 0 ? left : right;
	}
	constexpr const float &operator[](const size_t i) const noexcept
	{
		assert(i < 2);
		return i == 0 ? left : right;
	}
```

Reply:
Sure, sounds like a good idea.

Reply:
Done (but I couldn't call the non-const version as the resulted in infinite recursion).